app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -1,6 +1,27 @@
 module MfaSetupConcern
   extend ActiveSupport::Concern
 
+  def user_next_authentication_setup_path!(final_path = nil)
+    user_session[:current_mfa_created] = user_session[:current_mfa_created].nil? ?
+      0 : user_session[:current_mfa_created] + 1
```

Comment:
I think we need to initialize this somewhere else, since I just happened to stumble into an odd case where, if the user decides to revert back to the selection screen, the `:current_mfa_created` won't be reset, so it'll start from the _second_ of the new selected options (and there's an error page if there wasn't a second selected option).

I imagine it should probably be set to `0` when initially submitting the selection ? (`process_valid_form`)

Reply:
Now that I've taken a closer look at the logic here, I don't know that I mind the previous idea of the `Array#shift` behavior for iterating through the remaining selections, if it makes this sort of thing easier to manage. As long as the user can still refresh the page and such without the `shift`-ing eating their progress unexpectedly.

Reply:
It may be a little YAGNI, but session information is kind of a data migration with more complex forward/backward compatibility. People do sometimes jump back and forward multiple pages, so my hope was that not mutating the original selections would give us some more flexibility if we wanted to make changes to the flow in the future, or even show a step indicator of methods added.

The other option I had in mind was to use query parameters instead. Store all of the selected options, and then redirect to `option=0`, `option=1`, etc.

Reply:
Yea. Im thinking of adding to query params in a followup ticket. this ticket is under a feature toggle so shouldn't affect current functionality.